src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -88,56 +87,81 @@ def load_generic(config, interface_section)
         config.name = name_from_section(interface_section)
         config.interface = config.name # in autoyast name and interface is same
         if config.bootproto == BootProtocol::STATIC
-          if !interface_section.ipaddr
+          # we need at least some ip for the static configuration
+          if !interface_section.ipaddr && !interfaces_section.aliases
             msg = "Configuration for #{config.name} is invalid #{interface_section.inspect}"
             raise ArgumentError, msg
           end
 
-          if !interface_section.ipaddr.empty?
-            ipaddr = IPAddress.from_string(interface_section.ipaddr)
-          end
-          # Assign first netmask, as prefixlen has precedence so it will overwrite it
-          ipaddr.netmask = interface_section.netmask if !interface_section.netmask.to_s.empty?
-          if !interface_section.prefixlen.to_s.empty?
-            ipaddr.prefix = interface_section.prefixlen.to_i
-          end
-          if !interface_section.broadcast.empty?
-            broadcast = IPAddress.new(interface_section.broadcast)
-          end
-          if !interface_section.remote_ipaddr.empty?
-            remote = IPAddress.new(interface_section.remote_ipaddr)
-          end
-          config.ip = ConnectionConfig::IPConfig.new(
-            ipaddr, broadcast: broadcast, remote_address: remote
-          )
+          config.ip = load_ipaddr(interface_section)
         end
 
         # handle aliases
-        interface_section.aliases.each_value do |alias_h|
-          ipaddr = IPAddress.from_string(alias_h["IPADDR"])
-          # Assign first netmask, as prefixlen has precedence so it will overwrite it
-          ipaddr.netmask = alias_h["NETMASK"] if alias_h["NETMASK"]
-          ipaddr.prefix = alias_h["PREFIXLEN"].delete("/").to_i if alias_h["PREFIXLEN"]
-          config.ip_aliases << ConnectionConfig::IPConfig.new(ipaddr, label: alias_h["LABEL"])
+        interface_section.aliases.values.each_with_index do |alias_h, index|
+          config.ip_aliases << load_alias(alias_h, id: "_#{index}")
         end
+
+        # startmode
```

Comment:
isn't it self explanatory?, looks like useless comments

Reply:
well ... it was for me when organizing the work ... if it hurts, i can delete it